src/pybind/mgr/cephadm/module.py

```diff
@@ -931,17 +933,54 @@ def blink(host, dev):
 
         return blink(locs)
 
-    @async_completion
-    def _create_osd(self, all_hosts_, drive_group):
-        all_hosts = orchestrator.InventoryNode.get_host_names(all_hosts_)
-        assert len(drive_group.hosts(all_hosts)) == 1
-        assert len(drive_group.data_devices.paths) > 0
-        assert all(map(lambda p: isinstance(p, six.string_types),
-            drive_group.data_devices.paths))
-
-        host = drive_group.hosts(all_hosts)[0]
-        self._require_hosts(host)
+    def call_inventory(self, hosts, drive_groups):
+        def call_create(inventory):
+            return self._prepare_deployment(hosts, drive_groups, inventory)
+
+        return self.get_inventory().then(call_create)
+
+    def create_osds(self, drive_groups):
```

Comment:
The base class method should be changed if we want to accept multiple DGs in this call:

https://github.com/ceph/ceph/blob/9fc9197db00203865d43dc5117626a7b29358f1e/src/pybind/mgr/orchestrator.py#L871

And also existing implementations in `ansible`/`rook`/`test_orchestrator` module.